File: src/main/java/com/tms/csp/ast/formula/FConstraintSet.java
package com.tms.csp.ast.formula;


import com.tms.csp.ast.Exp;
import com.tms.csp.ast.Var;

/**
 * represents a set of complex fact
 */
public interface FConstraintSet extends Iterable<Exp> {

    /**
     * Does not include simple fact
     */
    int getConstraintCount();

    boolean isDirectlyRelated(int c1, int c2);
}

File: src/main/java/com/tms/csp/ast/formula/FConstraintSet.kt
package com.tms.csp.ast.formula


import com.tms.csp.ast.Exp

/**
 * represents a set of complex fact
 */
interface FConstraintSet  {

    /**
     * Does not include simple fact
     */
    val constraintCount: Int

    fun isDirectlyRelated(c1: Int, c2: Int): Boolean

    val argIt: Iterable<Exp>

}
